### ISSUE TITLE
Frame.Navigate performance problems with NavigationCacheMode.Required

### ISSUE THREAD

**Issue by FrayxRulez** [microsoft/microsoft-ui-xaml#2707]
**Describe the bug**
I have a chat app where UI is split in two: chat list on left side, open chat on right side.
Chat page has `NavigationCacheMode` set to `Required` as it is a critical part of the app and should be always accessed the fast as possible.
Clicking on different chats will cause a navigation to the chat page, triggering OnNavigatedTo, where I simply replace the current `DataContext` with a new one, initializing it, and here's the bottleneck: the navigation events seem to cause more work than they should, and the UI becomes unresponsive for a few instants, slowing down the user experience.
I am sure this is caused by `Frame.Navigate` as I tried to replace it by invoking the same methods triggered by `OnNavigatedTo` manually (by casting `Frame.Content` to my page type) and this way the navigation is instantaneous.

I am removing the rest of the bug template, as this issue is applicable to UWP itself, and likely on any version of it.

**Comment by StephenLPeters** [comment]
@MikeHillberg do you know who own the frame navigation code paths so we can inform the right owner?

**Comment by Noemata** [comment]
@FrayxRulez , you have to pre-cache pages you want to navigate to.  There is a one time hit when using this.NavigationCacheMode = NavigationCacheMode.Enabled,

We do this on load for some of the pages with a larger payload.

FYI: NavigationCacheMode.Required doesn't work.  Use NavigationCacheMode.Enabled instead.

**Comment by Noemata** [comment]
@FrayxRulez , the landscape on this has shifted since I first looked at this a few months back in a different context.  There are some tooling surprises that need workarounds.  The problems that existed at the start of 2019 have been fixed.  Now we have some new issues to contend with.

Look at the samples at the bottom of this page for ideas:

https://github.com/microsoft/microsoft-ui-xaml/issues/4713
